### ISSUE TITLE
ACR Tasks `task.yaml` expansion evaluation for debugging

### ISSUE THREAD

**Issue by SteveLasker** [Azure/acr#482]
**What is the problem you're trying to solve**
When debugging tasks, it's very tedious to debug the results of go templating. 
When errors occur, the line numbers don't reflect the numbers of the `tasks.yaml` authored, rather the expanded line numbers which can send you to very different areas.
There are additional challenges with `-` in `.` notation, or other character escaping.

**Describe the solution you'd like**
- A way to see the expanded go template, with line numbers that match the actual error line numbers.
- The ability to get the results of the yaml expansion without the actual execution `--dry-run` or some such equivalent

**Additional context**

**Comment by sajayantony** [comment]
Ideally it would be good if we could just spit out the expanded yaml on the client. The issue here is that there are secrets. Overall you are hitting the parts that need improvements for faster iteration.

**Comment by SteveLasker** [comment]
Yup, faster iteration (aka debugging) is definitely the core issue. Queued up for next semester
